pycfmodel/utils.py

```diff
@@ -14,3 +15,13 @@ def is_conditional_dict(value: Any) -> bool:
                 return False
         return True
     return False
+
+
+def regex_from_cf_string(action: str) -> Pattern:
+    # Replace *
+    action = action.replace("*", ".*")
+
+    # Replace ?
+    action = action.replace("?", ".{1}")
+
+    return re.compile(f"^{action}$", re.IGNORECASE)
```

Comment:
What do you think about rewriting this function using fnmatch now that is moved? https://docs.python.org/3/library/fnmatch.html

Reply:
Reading the docs: https://docs.aws.amazon.com/IAM/latest/UserGuide/reference_policies_elements_resource.html, it doesn't look like all fnmatch replacements are needed. Specially ranges with square brackets